src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx

```diff
@@ -39,18 +40,21 @@ export const MSG = defineMessages({
   },
 });
 
-const endDateTypes = [
+const endDateOptions: DatePickerOption[] = [
   {
     label: MSG.whenCancelled,
     value: ExpenditureEndDateTypes.WhenCancelled,
+    hideDatePicker: true,
   },
   {
     label: MSG.limitIsReached,
     value: ExpenditureEndDateTypes.LimitIsReached,
+    hideDatePicker: true,
```

Comment:
Wouldn't it be better to have this as toggle on, i.e. showDatePicker: true, then default, should be not having to include it.